composer/metrics/nlp.py

```diff
@@ -196,6 +196,37 @@ def compute(self) -> Tensor:
 
 class InContextLearningMetric(Metric):
 
+    def __init__(self, dist_sync_on_step=False, cache_responses=False):
+        super().__init__(dist_sync_on_step=dist_sync_on_step)
+        self.add_state('response_cache', default=[], dist_reduce_fx=None)
+        self.cache_responses = cache_responses
+
+    def reset_response_cache(self, cache: bool):
+        self.cache_responses = cache
+        setattr(self, 'response_cache', [])
+
+    def format_response_cache(self, tokenizer):
+        columns, rows = None, None
+        assert isinstance(self.response_cache, list)
+        if self.cache_responses and len(self.response_cache) > 0:
+            rows = []
+            for row in self.response_cache:
+                assert isinstance(row, dict)
+                columns = list(row.keys())
+                converted_row = []
+                for r_i in row.values():
+                    if isinstance(r_i, list) and len(r_i) > 0 \
+                        and all(isinstance(r_ij, int) for r_ij in r_i) \
+                        and not all(isinstance(r_ij, bool) for r_ij in r_i):
```

Comment:
I don't understand why we need all this type checking.

Reply:
see prev comment re pyright